frontend/src/components/pages/your_clubs/YourClubs.tsx

```diff
@@ -61,19 +61,38 @@ export const YourClubs = () => {
   /** Re-renders Profile only when number of displayed clubs changes. */
   React.useEffect(() => {
     (async() => {
-      let numClubsToDisplay = numClubsDisplayed 
-        ? numClubsDisplayed 
-        : DEFAULT_NUM_DISPLAYED;
-      const listedClubsPromise =
-        await yourClubsHandlerService.listClubs(numClubsToDisplay);
-      setListedClubs(listedClubsPromise);
+      const numClubsToDisplay = numClubsDisplayed
+        ? setNumClubsDisplayed(numClubsDisplayed)
+        : setNumClubsDisplayed(DEFAULT_NUM_DISPLAYED)
+      updateClubList();
     })();
   }, [numClubsDisplayed]);
 
   const handleNumClubsChange = (event: React.ChangeEvent<HTMLInputElement>) => {
     setNumClubsDisplayed(Number(event.target.value));
   }
 
+  const updateClubList = () => {
+    (async() => {
+      const listedClubsPromise =
+        await yourClubsHandlerService.listClubs(numClubsDisplayed);
+      setListedClubs(listedClubsPromise);
+    })();
+  }
```

Comment:
Can you do this instead?

```suggestion
  const updateClubList = async () => {
      const listedClubsPromise =
        await yourClubsHandlerService.listClubs(numClubsDisplayed);
      setListedClubs(listedClubsPromise);
  };
```